src/main/java/org/humancellatlas/ingest/file/web/FileEventHandler.java

```diff
@@ -0,0 +1,32 @@
+package org.humancellatlas.ingest.file.web;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.humancellatlas.ingest.core.service.ValidationStateChangeService;
+import org.humancellatlas.ingest.file.File;
+import org.humancellatlas.ingest.process.Process;
+import org.humancellatlas.ingest.state.ValidationState;
+import org.springframework.data.rest.core.annotation.HandleAfterLinkDelete;
+import org.springframework.data.rest.core.annotation.HandleAfterLinkSave;
+import org.springframework.data.rest.core.annotation.RepositoryEventHandler;
+
+import java.util.Set;
+
+import static org.humancellatlas.ingest.core.EntityType.FILE;
+
+@RepositoryEventHandler()
+@RequiredArgsConstructor
+public class FileEventHandler {
+
+    private final @NonNull ValidationStateChangeService validationStateChangeService;
+
+    @HandleAfterLinkDelete
+    public void handleProcessLinkingAfterDelete(File file, Set<Process> processes) {
+        validationStateChangeService.changeValidationState(FILE, file.getId(), ValidationState.DRAFT);
```

Comment:
You could extract a function and use in the 2 event handers

Reply:
I agree, I was thinking which file I should extract that function to, given that all three event handlers are calling the same method with different arguments. Do you have any suggestions?

Reply:
I think all of them could be in the same file.
See in https://www.baeldung.com/spring-data-rest-events the example with the handlers for "book" and "author" at the same file.

Reply:
done